lib/studies/reply/inbox.dart

```diff
@@ -76,56 +81,177 @@ class _BuildDesktopNav extends StatefulWidget {
-      _isExtended = widget.extended;
+      onLogoTapped();
+      _hasWidgetUpdated = true;
     }
   }
 
+  @override
+  void dispose() {
+    super.dispose();
+    _controller.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+
     return Scaffold(
       body: Row(
         children: [
           NavigationRail(
             destinations: [
               for (var destination in widget.destinations.keys)
                 NavigationRailDestination(
-                  icon: Icon(widget.destinations[destination]),
+                  icon: ImageIcon(
+                    AssetImage(
+                      widget.destinations[destination],
+                      package: _assetsPackage,
+                    ),
+                  ),
                   label: Text(destination),
                 ),
             ],
             extended: _isExtended,
             labelType: NavigationRailLabelType.none,
-            leading: Row(
+            leading: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
               children: [
-                const SizedBox(width: 25),
-                InkWell(
-                  child: const Icon(Icons.menu),
-                  onTap: () {
-                    setState(() {
-                      _isExtended = !_isExtended;
-                    });
-                  },
+                SizedBox(
+                  height: 56,
+                  child: Row(
+                    children: [
+                      const SizedBox(width: 12),
+                      SizedBox(
+                        width: _isExtended ? 120 : 58,
+                        child: InkWell(
+                          borderRadius: const BorderRadius.all(
+                            Radius.circular(16),
+                          ),
+                          child: Row(
+                            children: [
+                              RotationTransition(
+                                turns: Tween(
+                                  begin: _isExtended ? 0.0 : 0.5,
+                                  end: 1.0,
+                                ).animate(_controller.view),
+                                child: const Icon(
+                                  Icons.arrow_left,
+                                  color: ReplyColors.blue50,
+                                  size: 16,
+                                ),
+                              ),
+                              const ImageIcon(
+                                AssetImage(
+                                  'reply/reply_logo.png',
+                                  package: _assetsPackage,
+                                ),
+                                size: 32,
+                                color: ReplyColors.blue50,
+                              ),
+                              const SizedBox(width: 10),
+                              Text(
```

Comment:
would it be better ti wrap this Text in `if (_isExtended)` rather than using the empty string?

Reply:
I agree, I'll update it!